### PR TITLE
[SPARK-40260][SQL] Use error classes in the compilation errors of GROUP BY a position

### PR DESCRIPTION
**Pull request by MaxGekk**
### What changes were proposed in this pull request?
In the PR, I propose to the following new error classes:
- GROUP_BY_POS_OUT_OF_RANGE
- GROUP_BY_POS_REFERS_AGG_EXPR

and migrate 2 compilation exceptions related to GROUP BY a position onto them.

### Why are the changes needed?
The migration onto error classes makes the errors searchable in docs, and allows to edit error's text messages w/o modifying the source code.

### Does this PR introduce _any_ user-facing change?
Yes, in some sense because it modifies user-facing error messages.

### How was this patch tested?
By running the affected test suites:
```
$ build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite"
$ build/sbt "core/testOnly *SparkThrowableSuite"
```

### CODE REVIEW

**Comment by MaxGekk** [comment]
cc @srielau @anchovYu

**Review by cloud-fan — APPROVED** [review]

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -136,6 +136,18 @@
...
   },
+  "GROUP_BY_POS_OUT_OF_RANGE" : {
+    "message" : [
+      "GROUP BY position <index> is not in select list (valid range is [1, <size>])."
+    ],
+    "sqlState" : "42000"
+  },
+  "GROUP_BY_POS_REFERS_AGG_EXPR" : {
+    "message" : [
+      "GROUP BY <index> refers to an expression that is or contains an aggregate function. Aggregate functions are not allowed in GROUP BY, but got <aggExpr>."
```
Comment (srielau) [inline core/src/main/resources/error/error-classes.json:147]: How about this? Same content, just less wordy.

```suggestion
      "GROUP BY <index> refers to an expression <aggExpr> that contains an aggregate function. Aggregate functions are not allowed in GROUP BY."
```

**Review by srielau — APPROVED** [review]
Nit on the wording, LGTM otherwise

**Comment by MaxGekk** [comment]
Merging to master. Thank you, @srielau @cloud-fan for review.

**Pull request closed without merging**